rlclientlib/serialization/fb_serializer.h

```diff
@@ -273,6 +275,59 @@ namespace reinforcement_learning { namespace logger {
     }
   };
 
+  template <>
+  struct fb_collection_serializer<generic_event>
+  {
+    using serializer_t = fb_event_serializer<generic_event>;
+    using buffer_t = utility::data_buffer;
+    static int message_id() { return message_type::fb_generic_event_collection;; }
+
+    fb_collection_serializer(buffer_t &buffer, content_encoding_enum content_encoding)
+        : _allocator(buffer), _builder(buffer.body_capacity(), &_allocator), _buffer(buffer), _content_encoding(content_encoding) {}
+
+    int add(generic_event &evt, api_status *status = nullptr)
+    {
+      flatbuffers::Offset<typename serializer_t::fb_event_t> offset;
+      RETURN_IF_FAIL(serializer_t::serialize(evt, _builder, offset, status));
+      _event_offsets.push_back(offset);
+      return error_code::success;
+    }
+
+    uint64_t size() const { return _builder.GetSize(); }
+
+    const char * getTextForEnum(content_encoding_enum enumVal)
```

Comment:
This should be another function in context_encoding.h

Reply:
This will be entirely removed. I'm changing content_encoding to string instead of enum.